subsys/net/lib/lwm2m: Change how we seed the random generator

aescolar opened the pull request:
Instead of seeding the random generator from the test
itself calling into a host API, let's use the
entropy generator option to be seeded from /dev/urandom

This avoids trouble with the srandom() and time() calls
not existing depending on the chosen C library.

---

Sits on top of:
- https://github.com/zephyrproject-rtos/zephyr/pull/67998

---

Just as a sidenote. To pass the option, it would be also possible to do this below, but then you would only get the option if you build with twister. So it seemed nicer to just add it to the board config overlay.

```diff
diff --git a/tests/net/lib/lwm2m/interop/testcase.yaml b/tests/net/lib/lwm2m/interop/testcase.yaml
index 0639047226..266a8aa824 100644
--- a/tests/net/lib/lwm2m/interop/testcase.yaml
+++ b/tests/net/lib/lwm2m/interop/testcase.yaml
@@ -12,6 +12,8 @@ tests:
       - native_sim
       - qemu_cortex_m3
       - qemu_x86
+    extra_configs:
+      - arch:posix:CONFIG_NATIVE_EXTRA_CMDLINE_ARGS="--seed-random"
     tags:
       - testing
       - pytest
```

aescolar committed:
subsys/net/lib/lwm2m: Change how we seed the random generator

Instead of seeding the random generator from the test
itself calling into a host API, let's use the
entropy generator option to be seeded from /dev/urandom

This avoids trouble with the srandom() and time() calls
not existing depending on the chosen C library.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

SeppoTakalo reviewed (changes requested):
No, don't remove this.

This is a Interoperability test against a live server.
It MUST be a random, or else it might end up rotating same port numbers, same CoAP sequance numbers, same DTLS IDs.

Starting the same test twice, should allow it to pass.
Of there is no random at all, it would not pass the DTLS handshake.

SeppoTakalo commented:
Instead of `--seed=<seed_value>` I think we should add, either to command line, or to Twister, a functionality that allows me to seed it randomly.

I understand why it does not seed by default, but we should still allow `native_sim` tests against real world servers. It has proven to be extremely useful on testing LwM2M stack.

aescolar committed:
drivers entropy fake_native: Add option to seed generator randomly

Add a command line option which will seed the random generator
from /dev/urandom.
This can be usefull for some particular tests in which we are
interested in having different random numbers in each run,
but we cannot provide a different random seed from command line.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

SeppoTakalo reviewed (approved):
This looks good now.
Thanks for the new feature.
